app/scripts/ui.js

```diff
@@ -69,131 +76,50 @@ async function start() {
 
-  let isUIInitialised = false;
-
   // setup stream to background
   extensionPort = browser.runtime.connect({ name: windowType });
+
   let connectionStream = new PortStream(extensionPort);
 
   const activeTab = await queryCurrentActiveTab(windowType);
 
-  let loadPhishingWarningPage;
-
-  if (isManifestV3) {
-    /*
-     * In case of MV3 the issue of blank screen was very frequent, it is caused by UI initialising before background is ready to send state.
-     * Code below ensures that UI is rendered only after "CONNECTION_READY" or "startUISync"
-     * messages are received thus the background is ready, and ensures that streams and
-     * phishing warning page load only after the "startUISync" message is received.
-     * In case the UI is already rendered, only update the streams.
-     */
-    const messageListener = async (message) => {
-      if (message?.data?.method === 'startUISync') {
-        if (isUIInitialised) {
-          // Currently when service worker is revived we create new streams
-          // in later version we might try to improve it by reviving same streams.
-          updateUiStreams();
-        } else {
-          endTrace({ name: TraceName.BackgroundConnect });
-          initializeUiWithTab(activeTab);
-        }
-        await loadPhishingWarningPage();
-      }
-    };
-
-    /**
-     * An error thrown if the phishing warning page takes too long to load.
-     */
-    class PhishingWarningPageTimeoutError extends Error {
-      constructor() {
-        super('Timeout failed');
-      }
+  /*
+   * In case of MV3 the issue of blank screen was very frequent, it is caused by UI initialising before background is ready to send state.
+   * Code below ensures that UI is rendered only after "CONNECTION_READY" or "startUISync"
+   * messages are received thus the background is ready, and ensures that streams and
+   * phishing warning page load only after the "startUISync" message is received.
+   * In case the UI is already rendered, only update the streams.
+   */
+  const messageListener = async (message) => {
+    const method = message?.data?.method;
+
+    if (method !== METHOD_START_UI_SYNC) {
+      return;
     }
 
-    /**
-     * Load the phishing warning page temporarily to ensure the service
-     * worker has been registered, so that the warning page works offline.
-     */
-    loadPhishingWarningPage = async function () {
-      // Check session storage for whether we've already initialized the phishing warning
-      // service worker in this browser session and do not attempt to re-initialize if so.
-      const phishingSWMemoryFetch = await browser.storage.session.get(
-        PHISHING_WARNING_SW_STORAGE_KEY,
+    endTrace({ name: TraceName.BackgroundConnect });
+
+    if (isManifestV3 && isUIInitialised) {
+      // Currently when service worker is revived we create new streams
+      // in later version we might try to improve it by reviving same streams.
+      updateUiStreams();
```

Comment:
My understanding is that we can now be certain the service worker will never restart while the browser is open, meaning this logic and the `resetExtensionStreamAndListeners` branch below is superfluous.

But for the sake of keeping this PR as a pure refactor with no functional changes, I've left it for now.